pkg/codegen/internal/test/testdata/provider-config-schema/nodejs/config/vars.ts

```diff
@@ -5,18 +5,68 @@ import * as pulumi from "@pulumi/pulumi";
 import { input as inputs, output as outputs } from "../types";
 import * as utilities from "../utilities";
 
+declare var exports: any;
 let __config = new pulumi.Config("configstation");
 
-export let favoritePlants: string[] | undefined = __config.getObject<string[]>("favoritePlants");
+export declare const favoritePlants: string[] | undefined;
+Object.defineProperty(exports, "favoritePlants", {
+    get() {
+        return __config.getObject<string[]>("favoritePlants");
+    },
+    enumerable: true,
+});
+
 /**
  * omg my favorite sandwich
  */
-export let favoriteSandwich: outputs.config.Sandwich | undefined = __config.getObject<outputs.config.Sandwich>("favoriteSandwich");
-export let isMember: boolean | undefined = __config.getObject<boolean>("isMember") || true;
-export let kids: outputs.Child | undefined = __config.getObject<outputs.Child>("kids");
-export let name: string | undefined = __config.get("name");
-export let numberOfSheep: number | undefined = __config.getObject<number>("numberOfSheep");
+export declare const favoriteSandwich: outputs.config.Sandwich | undefined;
+Object.defineProperty(exports, "favoriteSandwich", {
+    get() {
+        return __config.getObject<outputs.config.Sandwich>("favoriteSandwich");
+    },
+    enumerable: true,
+});
+
+export declare const isMember: boolean | undefined;
```

Comment:
Unrelated to your change (and not something we have to do as part of this change), but I see that this is never going to be `undefined` because of the default value. Maybe file an issue to consider fixing the type in such cases?

Reply:
I'll just fix it here

Reply:
I wasn't referring to whether the property was required, I was referring to the default value.

`isMember`'s getter is:

```typescript
return __config.getObject<boolean>("isMember") ?? true;
```

Because of the `?? true` it will never return `undefined`.

`name`'s getter is:

```typescript
return __config.get("name");
```

It can still return `string | undefined` because `Config.get` returns `string | undefined`.

Reply:
Ah right okay fair enough, missed that about the default values but that's a fair point.

But `name` is required... so it won't ever actually return `undefined` - right? In [python](https://github.com/pulumi/pulumi/blob/master/pkg/codegen/internal/test/testdata/provider-config-schema/python/pulumi_configstation/config/__init__.pyi#L24) this is typed as `str` (not `Optional[str]`) - so I thought it should be the same here...

Reply:
> But `name` is required... so it won't ever actually return `undefined` - right?

It can return `undefined` if someone doesn't have that config value set.

Ah, I missed this in the python review. Since it's just calling `get` it should be `Optional[str]` not `str`. We should fix that. (Sorry, I didn't connect the dots w.r.t. `str` vs. `Optional[str]` in python until this discussion).

> Or should it be `__config.require()` instead?

It likely should, but that would be a behavior change that we should consider separately. (Not sure if we'd ever change it, to be honest).

In AWS, for Node.js, there is an overlay that adds require.ts that can be used to check:

https://github.com/pulumi/pulumi-aws/blob/master/sdk/nodejs/config/require.ts

Reply:
Got it. I thought the required checking was not an overlay and a byproduct of setting the config var as required. Okay thanks will fix up!